### ISSUE TITLE
Learning a mapping from sequence to a tree

### ISSUE THREAD

**Issue by usptact** [JohnLangford/vowpal_wabbit#1057]
I have a problem of learning a model that maps a sequence to a tree.

The input is a collection of relatively short sequences (less than 20 symbols each). Each symbol of the sequence carries a label as shown in this example (sequence of 7 symbols):
`
A B C C B D E
`
The total number of input symbol labels is finite and is known from the training set. In my example there are 5 possible symbol labels - A, B, C, D and E.

The goal is to output a tree that must have the nodes with the labels from input sequence + some extra nodes with tree-specific labels. The labels of those extra nodes are mentioned at least once in training set trees. The output should be something like this:

```
A 1
B 2
C 2
C 4
B 3
D 3
E 3
1 0
2 0
3 4
4 0
```

The other nodes that are not from input sequence are denoted with digits. The first column is the node; the second column is the parent node. The root node is 0.

Every label from the input sequence must appear exactly once in the output tree as a label of a node.

The features that I have access to are coming from the sequence only. Basically, I can extract a rich set of features from the sequence for each symbol. As for the output tree, I don't know how the tree structure looks like nor how many nodes there are in total (at least number of symbols from input + root node).

The training data consists of pairs: sequence and the corresponding tree.

The task that I am solving is not dependency parsing but the nature of the problem seems similar to me. Can I use the option `--search_task dep_parser` or it is specific to dependency parsing only?

As a bonus: a sequence in my problem can map to more than one tree (some are just more likely than others). I leave this as a different question.

**Comment by JohnLangford** [comment]
@KaiWeiChang can you comment?  This looks very close to dependency parsing to me so at most a small adaptation would be needed.

**Comment by JohnLangford** [comment]
I'm closing this as it's cold.  I still believe you could handle the problem in this way.
